### PR TITLE
feat: enable create proof binding

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR enables proof generation from Rust circuits that are exact same with ones we have tested from c++ side!

### CODE REVIEW

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Comment by TomTaehoonKim** [comment]
Is [a9e90c7](https://github.com/kroma-network/tachyon/pull/249/commits/a9e90c72d0f5236634f5381c055c9d249f2150b0) a fix commit?

File: vendors/halo2/src/circuits/simple_circuit.rs [hunk 1]
```diff
@@ -301,3 +301,112 @@ impl<F: FieldExt> Circuit<F> for SimpleCircuit<F> {
...
+
+            let mut proof = transcript.finalize();
+            let proof_last = prover.finalize_transcript();
+            proof.extend_from_slice(&proof_last);
+            proof
+        };
+        assert_eq!(halo2_proof, tachyon_proof);
+        // ANCHOR_END: test-circuit
+    }
+}
```
Comment (TomTaehoonKim) [inline vendors/halo2/src/circuits/simple_circuit.rs:412]: Can you please separate this change from this commit or fix the commit message?

**Comment by chokobole** [comment]
> Is [a9e90c7](https://github.com/kroma-network/tachyon/pull/249/commits/a9e90c72d0f5236634f5381c055c9d249f2150b0) a fix commit?

I think so. Because function in header file should be marked with `inline`, `constexpr` or `ALWYAS_INLINE`, otherwise it is not built when different .cc try to import header while causing duplicated symbol error.

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Pull request merged**